vmdb/app/models/orchestration_template.rb

```diff
@@ -60,6 +60,20 @@ def eligible_managers
     self.class.eligible_managers
   end
 
+  # return the validation error message; otherwise nil
+  def validate_content(manager = nil)
+    test_managers = manager.nil? ? eligible_managers : [manager]
+    test_managers.each do |mgr|
+      begin
+        return mgr.orchestration_template_validate(self)
+      rescue
+        # try next manager
+      end
+    end
+    # This manage may be shown in UI. Should we use term provider or manager?
+    "No manager is capable to validate the template"
```

Comment:
You can use the ui_lookup method here: `"No #{ui_lookup(:model => "ExtManagementSystem")} is capable of validating the template"`

```
ui_lookup(:model => "ExtManagementSystem")
=> "Provider"`
```
Defaults are defined here: vmdb/config/locales/en.yml